Sources/Subs-Members.php

```diff
@@ -513,22 +513,24 @@ function registerMember(&$regOptions, $return_errors = false)
 	if (!empty($regOptions['check_email_ban']))
 		isBannedEmail($regOptions['email'], 'cannot_register', $txt['ban_register_prohibited']);
 
-	// Check if the email address is in use.
+	// Check if the email address or the username is in use.
 	$request = $smcFunc['db_query']('', '
-		SELECT id_member
+		SELECT member_name
 		FROM {db_prefix}members
-		WHERE email_address = {string:email_address}
-			OR email_address = {string:username}
+		WHERE email_address = {string:email_address} OR member_name = {string:username}
```

Comment:
get this back the old stylew and change only the member_name column

Reply:
Done